go/mysql/constants.go

```diff
@@ -677,3 +678,33 @@ func IsSchemaApplyError(err error) bool {
 	}
 	return false
 }
+
+type ReplicationState int
+
+const (
+	ReplicationStateRunning ReplicationState = iota
+	ReplicationStateStopped
+	ReplicationStateConnecting
+	ReplicationStateUnknown
+)
+
+// ReplicationStatusToState converts a value you have for the IO thread(s) or SQL
+// thread(s) or Group Replication applier thread(s) from MySQL or intermediate
+// layers to a mysql.ReplicationState.
+// on,yes,true == ReplicationStateRunning
+// off,no,false == ReplicationStateStopped
+// connecting == ReplicationStateConnecting
+// anything else == ReplicationStateUnknown
+func ReplicationStatusToState(s interface{}) ReplicationState {
```

Comment:
IMHO the function should accept a `string`; it doesn't make sense for it to accept any type. It should be the caller's duty to provide a `string`.

Reply:
I don't feel strongly about it. I thought that sending a boolean would be nicer for some existing tests and uses. I can remove it though if there's good reason to.

Reply:
I'll do this now just to simplify things. Simplicity is itself a good reason on second thought. Thanks!